### PR TITLE
Updated BFGS to use Symmetric

### PR DESCRIPTION
**Pull request by btracey**

### CODE REVIEW

**Comment by btracey** [comment]
It's a bit slower at the moment, but it'll speed back up when gonum/matrix#100 is merged.

File: bfgs.go [hunk 1]
```diff
@@ -27,10 +27,13 @@ type BFGS struct {
...
-	y []float64
-	s []float64
+	y       []float64
+	yVec    *mat64.Vector
+	s       []float64
+	tmpData []float64
+	tmpVec  *mat64.Vector
 
-	invHess *mat64.Dense // TODO: Make symmetric when mat64 has symmetric matrices
+	invHess *mat64.SymDense // TODO: Make symmetric when mat64 has symmetric matrices
```
Comment (kortschak) [inline bfgs.go]: Remove TODO.

Comment (btracey) [inline bfgs.go]: Done. Sorry.

**Comment by kortschak** [comment]
LGTM

**Comment by btracey** [comment]
@jonlawlor this looks like the same error we were seeing earlier with Travis

**Comment by btracey** [comment]
Sorry, this is a real failure I guess. I think the partial pass was because I merged to matrix and here very close. Looking into it.

**Comment by btracey** [comment]
Nothing was wrong with this commit. The symmetric implementation in Inner is incorrect. Fixing now.

**Pull request merged**